src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -1677,35 +1677,64 @@ def get_placement_targets_by_zonegroup(self, zonegroup_name: str):
+                    if exit_code > 0:
+                        raise DashboardException(e=err,
+                                                 msg='Unable to add placement target {} to zonegroup {}'.format(placement_target['placement_id'], zonegroup_name),  # noqa E501  #pylint: disable=line-too-long
+                                                 http_status_code=500, component='rgw')
+                except SubprocessError as error:
+                    raise DashboardException(error, http_status_code=500, component='rgw')
+
+            if len(rgw_realm_list['realms']) < 1:
+               rgw_service_manager = RgwServiceManager()
+               rgw_service_manager.restart_rgw_daemons()
+            else:
+               self.update_period()
+            
             if storage_classes:
                 for sc in storage_classes:
-                    cmd_add_placement_options = ['--storage-class', sc]
-                    try:
-                        exit_code, _, err = mgr.send_rgwadmin_command(
-                            rgw_add_placement_cmd + cmd_add_placement_options)
-                        if exit_code > 0:
-                            raise DashboardException(e=err,
-                                                     msg='Unable to add placement target {} to zonegroup {}'.format(placement_target['placement_id'], zonegroup_name),  # noqa E501  #pylint: disable=line-too-long
-                                                     http_status_code=500, component='rgw')
-                    except SubprocessError as error:
-                        raise DashboardException(error, http_status_code=500, component='rgw')
-                    self.update_period()
+                    if sc == storage_class_name:
+                        cmd_add_placement_options = ['--storage-class', sc]
+                        try:
+                            exit_code, _, err = mgr.send_rgwadmin_command(
+                                                    rgw_add_placement_cmd + cmd_add_placement_options)
+                            if exit_code > 0:
+                                raise DashboardException(e=err,
+                                                         msg='Unable to add placement target {} to zonegroup {}'.format(placement_target['placement_id'], zonegroup_name),  # noqa E501  #pylint: disable=line-too-long
+                                                         http_status_code=500, component='rgw')
+                        except SubprocessError as error:
+                            raise DashboardException(error, http_status_code=500, component='rgw')
+                        if len(rgw_realm_list['realms']) < 1:
+                            rgw_service_manager = RgwServiceManager()
+                            rgw_service_manager.restart_rgw_daemons()
+                        else:
+                            self.update_period()
```

Comment:
this is being duplicated, maybe its own function?

Reply:
Agree +1